### PR TITLE
[10.0][IMP] account_payment_show_invoice: Add invoice on move line level

### PR DESCRIPTION
**Pull request by rousseldenis**
* Add invoice on move level
* Fix interface

### CODE REVIEW

**Comment by rousseldenis** [comment]
@lreficent What do you think of this ?
cc @acsone-odoo-team

**Review by Cedric-Pigeon — APPROVED** [review]
code review

File: account_payment_show_invoice/__manifest__.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "name": "Account Payment Show Invoice",
     "summary": "Extends the tree view of payments to show the paid invoices "
                "related to the payments using the vendor reference by default",
-    "version": "10.0.1.0.0",
+    "version": "10.0.1.1.0",
     "category": "Account-payment",
     "website": "https://odoo-community.org/",
     "author": "Eficent, Odoo Community Association (OCA)",
```
Comment (ThomasBinsfeld) [inline account_payment_show_invoice/__manifest__.py:11]: @rousseldenis The new field does not appear. You forgot to add the view file in the manifest.

**Review by ThomasBinsfeld — CHANGES_REQUESTED** [review]

File: account_payment_show_invoice/__manifest__.py (hunk 1 above)
Comment (rousseldenis) [reply account_payment_show_invoice/__manifest__.py:11]: Done

**Review by ThomasBinsfeld — APPROVED** [review]
Code and functional review :+1:

**Comment by Cedric-Pigeon** [comment]
@OCA/accounting-maintainers I think this one could be merged... Thanks

File: account_payment_show_invoice/views/account_move_line.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                </xpath>
+            </field>
+    </record>
+    <record id="view_move_line_tree" model="ir.ui.view">
+            <field name="name">account.move.line.tree (in account_payment_show_invoice)</field>
+            <field name="model">account.move.line</field>
+            <field name="inherit_id" ref="account.view_move_line_tree"/>
+            <field name="arch" type="xml">
+                <field name="move_id" position="after">
+                    <field name="payment_invoice_ids"/>
```
Comment (eLBati) [inline account_payment_show_invoice/views/account_move_line.xml]: @rousseldenis adding a not stored computed field to `account.move.line` list could be dangerous for performance

Comment (rousseldenis) [reply account_payment_show_invoice/views/account_move_line.xml]: @eLBati They are direct fields that are both stored (payment_id.invoice_ids). So, I think it should be acceptable (but depends on ORM - which has been improved on related fields)

File: account_payment_show_invoice/models/account_move_line.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class AccountMoveLine(models.Model):
+
+    _inherit = 'account.move.line'
+
+    payment_invoice_ids = fields.Many2many(
+        string="Payment Invoices",
+        related='payment_id.invoice_ids',
+        readonly=True,
```
Comment (eLBati) [inline account_payment_show_invoice/models/account_move_line.py:15]: @rousseldenis do you see problems with adding this
```suggestion
        readonly=True,
        store=True
```
?

**Comment by rousseldenis** [comment]
@lreficent @mreficent Concerning this module, there is a bug in interface if tree view is in editable bottom mode. invoice_ids field in account.payment tree should be set with widget "many2many_tags". This PR should be modified too. I will do the fix in this in another commit if you agree.

**Review by ThomasBinsfeld — APPROVED** [review]

File: account_payment_show_invoice/views/account_move_line.xml (hunk 2 above)
Comment (rousseldenis) [reply account_payment_show_invoice/views/account_move_line.xml]: @eLBati In production for months and no performance problem. Mutli-company, thousands of moves. Could you update your review?

Comment (eLBati) [reply account_payment_show_invoice/views/account_move_line.xml]: :+1:

**Comment by eLBati** [comment]
@rousseldenis I think the 2 commits should be squashed, as concerning the same change. Do you prefer to keep them separate?

Moreover, are these changes also applicable to v11?

Thanks

**Comment by rousseldenis** [comment]
@eLBati I'd prefer keeping them as it will facilitate feedback on that problem.

**Comment by eLBati** [comment]
@rousseldenis ok. What about v11?

**Comment by rousseldenis** [comment]
@eLBati Didn't test in v11. But I think this should be the case too.

**Pull request merged**